src/components/plots/abstractPlot.ts

```diff
@@ -429,6 +435,38 @@ export module Plot {
 
       return allSelections;
     }
+
+    /**
+     * Retrieves the closest PlotData to the specified x/y point within a specified value
+     *
+     * @param {number} xValue The x value to compare against
+     * @param {number} yValue The y value to compare against
+     * @param {number} withinValue The maximum distance the closest selection can be to the point (default = Infinity)
+     * @returns {PlotData} The closest plot data to the point within a specified value.  nulls and null selection returned otherwise
+     */
+    public getClosestData(xValue: number, yValue: number, withinValue: number = Infinity): PlotData {
+      var closestDatum: any = null;
+      var closestSelection = d3.select();
+      var closestPixelPoint: Point = null;
+      var closestPointDistance = withinValue;
+
+      this._getDrawersInOrder().forEach((drawer) => {
+        drawer._getRenderArea().selectAll(drawer._getSelector()).each(function() {
+          var selection = d3.select(this);
+          drawer._getPixelPoints(selection).forEach((pixelPoint: Point) => {
+            var pointDistance = Plottable._Util.Methods.pointDistance(pixelPoint, closestPixelPoint);
+            if (pointDistance < closestPointDistance) {
+              closestDatum = drawer._getDatum(selection, pixelPoint);
+              closestPixelPoint = pixelPoint;
+              closestSelection = selection;
+              closestPointDistance = pointDistance;
+            }
+          });
+        });
+      });
+
+      return { data: [closestDatum], pixelPoints: [closestPixelPoint], selection: d3.selectAll(closestSelection) };
```

Comment:
??? Why is the selection being re-selected? `closestSelection` is already a `D3.Selection`.